### PR TITLE
Asians

### PR DESCRIPTION
**Pull request by vargala4**

### CODE REVIEW

File: src/contract.py [hunk 1]
```diff
@@ -39,6 +39,9 @@ def to_dict(self) -> dict[str, any]:
             'observations': self.num_mon
         }
 
+    def set_num_mon(self, num_mon) -> list[float]:       # used by BB Barrier pricer
```
Comment (friedgab) [inline src/contract.py:42]: I would simply go with setting this property wherever you need. No need for this function.

File: src/contract.py [hunk 2]
```diff
@@ -208,30 +211,21 @@ def __init__(self, barrier_level: float, up_down: UpDown, in_out: InOut) -> None
...
-                return float(any([self._barrier_level <= price for price in observations]))
-            elif self._up_down == UpDown.DOWN:
-                return float(any([self._barrier_level >= price for price in observations]))
+        if np.isnan(vol):      # standard case without probabilities
+            if self.up_down == UpDown.UP:
+                return float(any([self.barrier_level <= price for price in observations]))
+            elif self.up_down == UpDown.DOWN:
+                return float(any([self.barrier_level >= price for price in observations]))
             else:
                 self._raise_incorrect_up_down_type()
```
Comment (friedgab) [inline src/contract.py:223]: No need for underscore prefix.

File: src/pricer.py [hunk 3]
```diff
@@ -591,29 +589,34 @@ def __init__(self, contract: EuropeanBarrierContract, model: MarketModel, params
...
         else:
             raise TypeError(f'MC is not supported for model type {type(contract).__name__}')
 
-    def calc_fair_value(self) -> float:
-            contract = self._contract
-            # num_mon_mod = round(contract._num_mon / 10)    # BB specific
-            num_mon_mod = contract._num_mon
+    def calc_fair_value_with_ci(self) -> float:
+            contract = self.contract
+            # num_mon_mod = round(contract.num_mon / 10)    # BB specific
```
Comment (friedgab) [inline src/pricer.py:600]: Could you please remove commented code?

File: src/contract.py (hunk 1 above)
Comment (friedgab) [inline src/contract.py:42]: I would simply go with setting this property whereever you need. There's no point of having this setter method.

File: src/contract.py (hunk 2 above)
Comment (vargala4) [reply src/contract.py:223]: deleted, also from the raise_unsupported_greek_method_error

File: src/pricer.py (hunk 3 above)
Comment (vargala4) [reply src/pricer.py:600]: deleted (however, would make sense to always scale down the nr of monitoring points for the BB pricer...)

File: src/contract.py (hunk 1 above)
Comment (friedgab) [reply src/contract.py]: I'm not against setting the number of monitoring points to a smaller number for barrier options. It's just useless to do it via a method in the current framework. Just use `contract.num_mon = number`.

**Review by friedgab — APPROVED** [review]

File: src/contract.py (hunk 1 above)
Comment (vargala4) [reply src/contract.py]: Indeed, but originally everything was with underscore.

**Pull request merged**